src/ruleset.rs

```diff
@@ -280,13 +250,13 @@ impl Ruleset {
                 handled_access_fs: self.actual_handled_fs.bits(),
             };
 
-            match self.compat.abi {
+            match self.compat.abi() {
                 ABI::Unsupported => {
                     #[cfg(test)]
-                    assert_eq!(self.compat.state, CompatState::Final);
+                    assert_eq!(self.compat.state(), CompatState::Final);
                     Ok(RulesetCreated::new(self, -1))
                 }
-                ABI::V1 => {
+                ABI::V1 | ABI::V2 => {
```

Comment:
It looks like you have this pattern 3 times in this file around a uapi call. I guess the idea is that more explicitness will be less error prone in the event that a syscall changes in the future?  My worry is that this list will grow long over a lot of ABI versions and could become more error prone that way.  If just matching `_` isn't amenable to you, maybe derive PartialOrd on ABI or something and match a range of versions instead of a full list?

I guess you understand better than me how this api might conceivably evolve in the future, so maybe I'm missing something here?

Reply:
You're right, this notation will not scale. Because there is no specific handling of the Landlock syscalls according to the Landlock ABI versions for now, I will use `_`. Implementing and using a range for `ABI` will be useful later though.